test/ExecutionTests.jl

```diff
@@ -148,12 +148,38 @@ tune!(b_pass)
 # warmup #
 ###########
 
-p = params(warmup(@benchmarkable sin(1)))
+@test_deprecated warmup(@benchmarkable sin(1))
+
+is_warm = false
+function needs_warm()
+    global is_warm
+    if is_warm
+        sleep(0.1)
+    else
+        sleep(2)
+        is_warm = true
+    end
+end
+
+w = @benchmarkable needs_warm()
+w.params.seconds = 1
+
+#test that all measurements from lineartrial used in tune! are warm
+is_warm = false
+@test maximum(BenchmarkTools.lineartrial(w, w.params)) < 1e9
+
+#test that run warms up the benchmark
+tune!(w)
+is_warm = false
+@test minimum(run(w).times) < 1e9
+
+#test that belapsed warms up the benchmark
```

Comment:
should there be tests for the other `@b...` macros, which were also modified?

Reply:
We can easily test `@benchmark`. It would be harder to test `@btime` or `@bprofile` because their output is harder to parse in the test suite.

Reply:
and @belapsed calls @benchmark, not sure which ones you'd like me to test.

Reply:
ok let's keep it that way